src/Tests/Microsoft.DotNet.ApiCompatibility.Tests/Rules/AttributesMustMatchTests.cs

```diff
@@ -666,12 +666,112 @@ public class First {
         [MemberData(nameof(MembersCases))]
         public void EnsureDiagnosticIsReported(string leftSyntax, string rightSyntax, CompatDifference[] want)
         {
+            using TempDirectory root = new();
+            string filePath = Path.Combine(root.DirPath, "exclusions.txt");
+            File.Create(filePath).Dispose();
+            TestRuleFactory s_ruleFactory = new((settings, context) => new AttributesMustMatch(settings, context, new[] { filePath }));
             IAssemblySymbol left = SymbolFactory.GetAssemblyFromSyntax(leftSyntax);
             IAssemblySymbol right = SymbolFactory.GetAssemblyFromSyntax(rightSyntax);
             ApiComparer differ = new(s_ruleFactory);
             IEnumerable<CompatDifference> got = differ.GetDifferences(new[] { left }, new[] { right });
             Assert.Equal(want, got);
         }
+
+        [Fact]
+        public void TestExclusionsFilteredOut()
+        {
+            using TempDirectory root = new();
+            string filePath = Path.Combine(root.DirPath, "exclusions.txt");
+            File.WriteAllText(filePath, "T:System.SerializableAttribute");
+            TestRuleFactory s_ruleFactory = new((settings, context) => new AttributesMustMatch(settings, context, new[] { filePath }));
+            string leftSyntax = @"
+namespace CompatTests
+{
+  using System;
+  
+  [AttributeUsage(AttributeTargets.All, AllowMultiple = true)]
+  internal class FooAttribute : Attribute {
+    public FooAttribute(String s) {}
+    public bool A;
+    public int B;
+  }
+
+  [Foo(""S"", A = true, B = 3)]
+  public class First {}
+}
+";
+            string rightSyntax = @"
+namespace CompatTests
+{
+  using System;
+  
+  [AttributeUsage(AttributeTargets.All, AllowMultiple = true)]
+  internal class FooAttribute : Attribute {
+    public FooAttribute(String s) {}
+    public bool A;
+    public int B;
+  }
+
+  [Serializable]
+  [Foo(""S"", A = true, B = 3)]
+  public class First {}
+}
+";
+            IAssemblySymbol left = SymbolFactory.GetAssemblyFromSyntax(leftSyntax);
+            IAssemblySymbol right = SymbolFactory.GetAssemblyFromSyntax(rightSyntax);
+            ApiComparer differ = new(s_ruleFactory);
+            IEnumerable<CompatDifference> got = differ.GetDifferences(new[] { left }, new[] { right });
+            Assert.Empty(got);
+        }
     }
 
+    internal class TempDirectory : IDisposable
```

Comment:
Please put that class into its own file and put it into src/Tests/Microsoft.DotNet.ApiCompatibility.Tests/TempDirectory.cs so that its clear that other tests can use it as well when needed.

Reply:
Also is that file coped from somewhere? I think we already have that one in runtime. Did you check if this file already exists in the sdk repo?